internal/storage/bucket_handle.go

```diff
@@ -210,14 +210,18 @@ func (b *bucketHandle) ListObjects(ctx context.Context, req *gcs.ListObjectsRequ
 	itr := b.bucket.Objects(ctx, query) // Returning iterator to the list of objects.
 	pi := itr.PageInfo()
 	pi.MaxSize = req.MaxResults
+	pi.Token = req.ContinuationToken
 	var list gcs.Listing
 
 	// Iterating through all the objects in the bucket and one by one adding them to the list.
 	for {
 		var attrs *storage.ObjectAttrs
+		// Fetching number of objects equals to requested MaxResults
```

Comment:
Please update the comment like this. itr.next returns all the objects present in the bucket. Hence adding a check to break after required number of objects are returned.

Reply:
fixed